tools/build_defs/repo/jvm.bzl

```diff
@@ -51,17 +51,18 @@ def _jvm_import_external(repository_ctx):
     name = repository_ctx.attr.generated_rule_name or repository_ctx.name
     urls = repository_ctx.attr.jar_urls
     sha = repository_ctx.attr.jar_sha256
-    path = repository_ctx.name + ".jar"
+    extension = "." + repository_ctx.attr.packaging
+    path = repository_ctx.name + extension
     for url in urls:
-        if url.endswith(".jar"):
+        if url.endswith(extension):
             path = url[url.rindex("/") + 1:]
             break
     srcurls = repository_ctx.attr.srcjar_urls
     srcsha = repository_ctx.attr.srcjar_sha256
-    srcpath = repository_ctx.name + "-src.jar" if srcurls else ""
+    srcpath = repository_ctx.name + "-src" + extension if srcurls else ""
```

Comment:
Packaging might be overloaded in this domain, related to maven, and I for one would be surprised if it would search for a sources file with extension zip for example.
Not sure how to name this attribute

Reply:
That's a good point. I'll revert it and leave it as it is.